### ISSUE TITLE
After a script load error when using useLoadScript it is not possible to load any maps on the page

### ISSUE THREAD

**Issue by coulson84** [JustFly1984/react-google-maps-api#282]
This is another quite specific offline issue that we have found when testing our offline functionality in our app. There is a workaround for it at the moment that can be achieved by passing in an ID for the script but I know that this is something (the passing of the ID) that has been talked about being removed when it comes to the next major version.

Essentially the problem is that when a script fails to load it is not removed from the page so when `injectScript` runs again it finds the previous script element which failed to load and attempts to do nothing. After a script fails to load it should probably be removed from the DOM so that it no longer affects any subsequent load requests.

If that all makes sense and you agree with it then I'll implement a fix and submit a PR

**Comment by JustFly1984** [comment]
@coulson84 I agree with you. Your PR is welcome.

**Comment by lluisrojass** [comment]
@coulson84 @JustFly1984 why not just extend the functionality of the data attribute already on the  script ? 

extract the script setup process to a separate function, add an error handler 

```js
script.onerror = function() { 
   script.setAttribute('data-state', 'error');
   reject()
}
```

**Comment by JustFly1984** [comment]
@lluisrojass your PR is welcome

**Comment by JustFly1984** [comment]
released in 1.7.14
